### PR TITLE
Allow adding and subtracting any amount with the zero value

### PR DESCRIPTION
**Pull request by josharian**
Updates #16

### CODE REVIEW

File: amount.go [hunk 1]
```diff
@@ -148,6 +148,12 @@ func (a Amount) Convert(currencyCode, rate string) (Amount, error) {
 
 // Add adds a and b together and returns the result.
 func (a Amount) Add(b Amount) (Amount, error) {
+	if a.Equal(Amount{}) {
```
Comment (josharian) [inline amount.go:151]: This does cause a 25% performance regression in the benchmark. If you preferred, at the cost of some readability, I could move these checks to instide the `if a.currencyCode != b.currencyCode` block at line 152.

Comment (bojanz) [reply amount.go]: Thank you, I think that was the right decision.

**Comment by bojanz** [comment]
Random thoughts:
```
			// 0-b == -b
			var result apd.Decimal
			result.Neg(&b.number)
			return Amount{result, b.currencyCode}, nil
```
It would probably make sense to add a Neg() method to Amount, then we could just return b.Neg(). I can do that in a followup.

Thank you for this work, and the upstream improvements you're making to cockroachdb/apd.

**Comment by josharian** [comment]
> It would probably make sense to add a Neg() method to Amount, then we could just return b.Neg(). I can do that in a followup.

Sounds great. Thanks!

> Thank you for this work, and the upstream improvements you're making to cockroachdb/apd.

My pleasure. Thanks for the reviews, and for doing the hard work of being a maintainer. I know it can be a lonely, tiring role.

I would love to also add some fuzz tests to this package, but first someone (maybe eventually me) needs to fix the apd issues that fuzzing finds all but immediately. :P

**Pull request merged**